From ee461ded1b24783c5b1ec1c53163d77be6d6cf52 Mon Sep 17 00:00:00 2001
From: Alexey Rochev <equeim@gmail.com>
Date: Wed, 09 Nov 2022 01:33:00 +0300
Subject: [PATCH] Core: always link with CMAKE_DL_LIBS

`dlopen` feature is used in qlibraryinfo.cpp without `library` feature.
Change condition that adds libdl library to make sure that we can build without `library` feature.

Pick-to: 6.4
Change-Id: I5051e6b2e0a8daae5f7e269ed8980df99ecaa192
---

diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
index f9dc9f7..0e90104 100644
--- a/src/corelib/CMakeLists.txt
+++ b/src/corelib/CMakeLists.txt
@@ -1091,7 +1091,7 @@
         plugin/qlibrary_unix.cpp
 )
 
-qt_internal_extend_target(Core CONDITION QT_FEATURE_dlopen AND QT_FEATURE_library
+qt_internal_extend_target(Core CONDITION QT_FEATURE_dlopen
     LIBRARIES
         ${CMAKE_DL_LIBS}
 )
